### PR TITLE
Add nx to package.json for npm run build to work

### PR DESCRIPTION
**Pull request by bjorn-einar-bjartnes-4ss**
# Problem
There seems to be issues with lerna when nx is not installed.

Running 
```
npm i
npm run build
``` 
fails with
```
❯ npm run build

> build
> lerna run build

lerna notice cli v6.0.3
lerna info versioning independent
lerna notice filter including ["msal","@azure/msal-common","@azure/msal-browser","@azure/msal-react","@azure/msal-angular","@azure/msal-node"]
lerna info filter [
lerna info filter   'msal',
lerna info filter   '@azure/msal-common',
lerna info filter   '@azure/msal-browser',
lerna info filter   '@azure/msal-react',
lerna info filter   '@azure/msal-angular',
lerna info filter   '@azure/msal-node'
lerna info filter ]
lerna ERR!
 There was a critical error when configuring the task runner, please report this on https://github.com/lerna/lerna
lerna ERR! Error: Cannot find module 'nx/src/utils/output'
lerna ERR! Require stack:
lerna ERR! - C:\code\microsoft-authentication-library-for-js\node_modules\@lerna\run\index.js
lerna ERR! - C:\code\microsoft-authentication-library-for-js\node_modules\@lerna\run\command.js
lerna ERR! - C:\code\microsoft-authentication-library-for-js\node_modules\lerna\index.js
lerna ERR! - C:\code\microsoft-authentication-library-for-js\node_modules\lerna\cli.js
lerna ERR!     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
lerna ERR!     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
lerna ERR!     at Module.require (node:internal/modules/cjs/loader:1005:19)
lerna ERR!     at require (node:internal/modules/cjs/helpers:102:18)
lerna ERR!     at RunCommand.configureNxOutput (C:\code\microsoft-authentication-library-for-js\node_modules\@lerna\run\index.js:350:24)
lerna ERR!     at RunCommand.runScriptsUsingNx (C:\code\microsoft-authentication-library-for-js\node_modules\@lerna\run\index.js:201:10)
lerna ERR! lerna Cannot find module 'nx/src/utils/output'
lerna ERR! lerna Require stack:
lerna ERR! lerna - C:\code\microsoft-authentication-library-for-js\node_modules\@lerna\run\index.js
lerna ERR! lerna - C:\code\microsoft-authentication-library-for-js\node_modules\@lerna\run\command.js
lerna ERR! lerna - C:\code\microsoft-authentication-library-for-js\node_modules\lerna\index.js
lerna ERR! lerna - C:\code\microsoft-authentication-library-for-js\node_modules\lerna\cli.js
```
image

# Added nx
```
npm install nx --save-dev
```
unfortunately rewrites a lot of package.lock, not sure how to make the changes have less of an impact.

Maybe another option is to include it in documentation as a dev dependency, to install nx globally or similar in https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/contributing.md#build-library

Or perhaps it is lerna that should have nx as a dependency? 😕

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "gh-pages": "^3.1.0",
     "lerna": "^6.0.3",
     "npm-run-all": "^4.1.5",
+    "nx": "^15.4.2",
```
Comment (bjorn-einar-bjartnes-4ss) [inline package.json:29]: I do not know why this caused massive changes to the package.lock file.

**Comment by bmahall** [comment]
@bjorn-einar-bjartnes-4ss Can you open an accompanying issue please to enable further discussion around this; and try installing npx using "npx add-nx-to-monorepo" & see if it leads to large package-lock changes?

**Comment by bjorn-einar-bjartnes-4ss** [comment]
Will do. 

The ```npx add-nx-to-monorepo``` I have to look into, I don't really know my way around these nx tools, so not sure what to answer to the questions about run order/cachable builds/distributed caching. Tried to go with the default selections as shown below, and it still made a huge diff to package-lock. I am not so sure it is a problem or not, but I did not want to suggest making all those changes to package deps just to get lerna to run. 

```
√ Which scripts need to be run in order?  (e.g. before building a project, dependent projects must be built.) · No items were selected
√ Which scripts are cacheable? (Produce the same output given the same input, e.g. build, test and lint usually are, serve and start are not) · No items were selected
√ Enable distributed caching to make your CI faster · Yes
```

**Comment by tnorling** [comment]
I haven't had this issue but this seems like something lerna should address. I see that they've listed `nx` as a devDependency, if it's required to run their commands they should list it either as a dependency or a peerDependency

**Comment by bjorn-einar-bjartnes-4ss** [comment]
I can raise it at lerna, then. I guess a lot of angular devs will have nx installed as a global tool. I don't know these toolchains very well. I guess until lerna has added nx as a proper dependency, it can be commented in the docs that it is required to make the build run.

**Comment by JamesHenry** [comment]
Hi Folks, I'm sorry you ran into this issue, we will take a look into it today.

@tnorling the root package.json of the lerna monorepo is not related to the packages themselves

The `lerna` package which gets published to npm does specify its dependency on nx: https://github.com/lerna/lerna/blob/99a2469c51d38dbe51c534487b44703041f36911/core/lerna/package.json#L66

**Comment by JamesHenry** [comment]
I have opened https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5533 which updates lerna to the latest and resolves this issue. Thanks all!

**Pull request closed without merging**